privacyidea/api/machine.py

```diff
@@ -206,6 +210,10 @@ def detach_token_api(serial, machineid, resolver, application):
          "application": "luks" }
 
     """
+    # In case of offline we are ignoring the machineid and the resolver.
+    if application.lower() == "offline":
+        machineid = None
+        resolver = None
```

Comment:
If we have an offline token still attached to a machine, it won't be deleted since it won't find the machine-token here:
https://github.com/privacyidea/privacyidea/blob/d1a5533f012090ef83f8d882c9f993ad557911e7/privacyidea/models.py#L2144
and here:
https://github.com/privacyidea/privacyidea/blob/d1a5533f012090ef83f8d882c9f993ad557911e7/privacyidea/lib/machine.py#L223
because of the missing `machine_id` and `machineresolver_id`

Reply:
So you mean if there is an "old token" with an attachement to a dedicated machine?

Reply:
I will write a test that does the following:

1. Create machine resolver
2. Create an HOTP token
3. Attach this token to a distinct machine
4. Authenticate
5. Detach this token from the offline application and machine

And then fix the code that it will work out properly. This way we will also have ensure a kind of migration scenario.

Reply:
I fixed this in 1d5945d9

The fix was actually to remove the Nullization in case of offline.
Since the "new" UI always uses "any machine" and "no resolver" to detach tokens.